Server/ChessArena.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+service ChessArena {
+    // A simple RPC
+    //
+    // Inform server that the client is ready to start the game. 
+    // Returns one of the following:
+    // 1. Player is starting player (white). Opponent move message not set.
+    // 2. Player is not starting player (black). Opponent move message provides opening
+    //    move coordinates. 
+    rpc Initialize(PlayerInformation) returns (GameStartInformation) {}
+
+    // A Bidirectional streaming RPC.
+    //
+    // Send player moves. Receive opponent moves.
+    rpc CreateMovements(stream PlayerMove) returns (stream OpponentMove) {}
+}
+
+// Initialize
+
+message PlayerInformation {
+    string name = 1;    
+}
+
+message GameStartInformation {
+    // True if client is starting player.
+    // If false, opponent move is also returned.
+    bool whitePlayer = 1;
+
+    MoveCoordinates opponentMove = 2;
+}
+
+
+// Create movements
+
+message PlayerMove {
+    MoveCoordinates coordinates = 1;
+
+    // Any optional additional data about the move.
+    // E.g. search depth, eval count, strategy...
+    string diagnostics = 2;
+}
+
+message OpponentMove {
```

Comment:
As the `OpponentMove` already includes information about the game (`gameEnd`), wouldn't "GameStatus" be more descriptive name? I think it should also include information if player's king is threatened (check).

It could include the board status as well as I think it is a little ambiguous to leave it for the AIs to keep track of the board.

Reply:
Or do you feel it's too much overhead?

Reply:
So I see two options here:

1. Server-side has some basic game intelligence. It tracks the current game state, knows if a move is illegal, capture, castling. Check and checkmate are probably easier to handle on AI side and send through `PlayerMove`. These can be further utilized on the web app elements (visualization etc).
2. Server-side only handles traffic between AIs. Board status is included in each move. Actions are only known by AIs. `PlayerMove` can be extended to give more parameters if needed.

Reply:
I think 1. would be a bit more safe bet. On 2. suppose there is a corrupted move from one AI. If other AI does not validate enough carefully, game would continue in faulty state. It would be noticed only by observing the match enough carefully.

Reply:
Latest commit has simplified `Move` now that contains all relevant info.